src/main/java/uk/gov/hmcts/cmc/claimstore/services/SettlementAgreementService.java

```diff
@@ -43,24 +44,46 @@ public SettlementAgreementService(
     }
 
     public Claim reject(Claim claim, String authorisation) {
-        assertSettlementIsNotReached(claim);
-        assertLastStatementIsClaimantAcceptation(claim);
-
-        MadeBy party = MadeBy.DEFENDANT;
 
-        Settlement settlement = claim.getSettlement()
-            .orElseThrow(() -> new ConflictException("Settlement agreement has not yet been made."));
+        Settlement settlement = assertSettlementCanBeResponded(claim);
+        settlement.reject(MadeBy.DEFENDANT);
 
-        settlement.reject(party);
 
-        String userAction = format("SETTLEMENT_AGREEMENT_REJECTED_BY_%s", party.name());
+        String userAction = format("SETTLEMENT_AGREEMENT_REJECTED_BY_%s", MadeBy.DEFENDANT.name());
         caseRepository.updateSettlement(claim, settlement, authorisation, userAction);
+
         Claim updated = claimService.getClaimByExternalId(claim.getExternalId(), authorisation);
+
         eventProducer.createSettlementAgreementRejectedEvent(updated);
         appInsights.trackEvent(SETTLEMENT_AGREEMENT_REJECTED, REFERENCE_NUMBER, updated.getReferenceNumber());
         return updated;
     }
 
+    public Claim countersign(Claim claim, String authorisation) {
+
+        Settlement settlement = assertSettlementCanBeResponded(claim);
+        settlement.countersign(MadeBy.DEFENDANT);
+
+        String userAction = format("SETTLEMENT_AGREEMENT_COUNTERSIGNED_BY_%s", MadeBy.DEFENDANT.name());
+        caseRepository.updateSettlement(claim, settlement, authorisation, userAction);
+
+        Claim updated = claimService.getClaimByExternalId(claim.getExternalId(), authorisation);
+
+        eventProducer.createSettlementAgreementCountersignedEvent(updated);
+        appInsights.trackEvent(SETTLEMENT_AGREEMENT_REACHED, REFERENCE_NUMBER, updated.getReferenceNumber());
+        return updated;
+
+    }
+
+    private Settlement assertSettlementCanBeResponded(Claim claim) {
```

Comment:
Looks weird that an `assertX` method returns something... can't think of a better name for it though.